strawberry/types/type_resolver.py

```diff
@@ -113,7 +113,7 @@ class if one is not set by either using an explicit strawberry.field(name=...) o
             # or the class where this field was defined, so we always have
             # the correct origin for determining field types when resolving
             # the types.
-            field.origin = field.origin or cls
+            field.origin = cls
```

Comment:
I couldn't figure out why the original code preferred `field.origin` over `cls`. I didn't find a any place where this is needed.

Reply:
I think the comment above this line may need some updating? We're always setting the field origin to the class now?

Reply:
Do you remember why you introduced that `field.origin or cls` in the first place?

Reply:
I believe it's a holdover from when we used to monkeypatch field information onto the resolver function instead of having dedicated `StrawberryField` and `StrawberryResolver` objects.

Reply:
I just realized that this fixed one use case but breaks another one: class inheritance

Previously a field would know it's origin and look up it's type relative to the origin. Now a subclass will set the origin of all fields, also the inherited ones breaking the lookup for those.

Until this is resolved this PR should not be merged.

Reply:
do we have a test for this? :)

Reply:
I seam to be wrong about this. I just added tests for this and my change doesn't break inheritance. So yes. Now there is a test for this and to my surprise it doesn't fail.

Reply:
Gotcha. That issue with inheritance does indeed arise when inheriting from a class that has some fields defined as `x: int = strawberry.field()`.

Reply:
This line was indeed correct so I revered the change. The culprit was in the `base_resolver` setter which did set the `origin`. Removing that line was enough to fix all test cases. :tada: